apps/src/webapp/webapp.js

```diff
@@ -208,24 +214,41 @@ Webapp.onTick = function() {
   }
 
   if (Webapp.interpreter) {
-    var doneUserCodeStep = false;
+    var doneUserLine = false;
+    var reachedBreak = false;
     var unwindingAfterStep = false;
     var inUserCode;
     var userCodeRow;
     var session = StudioApp.editor.aceEditor.getSession();
+    // NOTE: when running with no source visible or at max speed with blocks, we
+    // call a simple function to just get the line number, otherwise we call a
+    // function that also selects the code:
+    var selectCodeFunc =
+      (StudioApp.hideSource ||
+       (maxSpeed && !Webapp.paused && StudioApp.editor.currentlyUsingBlocks)) ?
```

Comment:
maxSpeed --> isMaxSpeed or atMaxSpeed, otherwise this reads like I'm testing if maxSpeed is zero here.